src/backend/app/projects/project_crud.py

```diff
@@ -54,15 +55,58 @@ async def create_project_with_project_info(
 
 
 async def get_project_by_id(
-    db: Session = Depends(database.get_db), project_id: Optional[int] = None
+    db: Database = Depends(database.encode_db), project_id: Optional[int] = None
 ) -> db_models.DbProject:
-    """Get a single project by id."""
-    db_project = (
-        db.query(db_models.DbProject)
-        .filter(db_models.DbProject.id == project_id)
-        .first()
-    )
-    return await convert_to_app_project(db_project)
+    """Get a single project &  all associated tasks by ID."""
+    #check the project in Database
+    raw_sql = f"""SELECT id, short_description,total_tasks FROM projects WHERE id = '{project_id}' LIMIT 1;"""
+    raw_sql = f"""
+    SELECT 
+        projects.id, 
+        projects.short_description, 
+        projects.description,
+        projects.author_id,
+        users.name AS author_name
+    FROM projects 
+    JOIN users ON projects.author_id = users.id 
+    WHERE projects.id = '{project_id}' 
+    LIMIT 1;
+    """
+
+    project =  await db.fetch_one(query=raw_sql)
+    if not project:
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Project not found.")
+
+    # get tasks of project
+    query = f""" SELECT * from tasks WHERE project_id = '{project_id}';"""
+    all_tasks = await db.fetch_all(query=query)    
+    tasks =   await convert_to_app_tasks(all_tasks)
+    # Count the total tasks
+    task_count = len(tasks)
+    return {
+        "project": project,
+        "task_count": task_count,
+        "tasks": tasks,
+    }
+    
+async def convert_to_app_tasks(
```

Comment:
This code is not necessary with Pydantic. The logic should be directly in the models instead of passed through a function afterwards 👍

FMTM actually needs this refactored out at some time too

Reply:
I have made changes to the code so that the logic is now directly in the Pydantic models. Additionally, I have removed the f-string from the SQL query. Let me know if there is anything else needed.